api/src/main/resources/workbench.yaml

```diff
@@ -2040,7 +2074,50 @@ definitions:
         description: >
           All the selected values in the data set
         items:
-          $ref: "#/definitions/ValueSet"
+          $ref: "#/definitions/DomainValuePair"
+
+  DomainValuePair:
+    type: object
+    properties:
+      domain:
+        description: >
+          Domain corresponding to an OMOP table that can contain rows for the concepts in this
+          concept set. Note that the Domain values RACE, GENDER, and ETHNICITY are not allowed
+          here; it makes sense to specify concepts in these domains in cohort criteria, but
+          there isn't much value in having concept sets defined for them.
+        $ref: "#/definitions/Domain"
+      value:
+        type: string
+
+  DataSetResponse:
```

Comment:
I think this could just be `DataSet`? Because it could be part of other response objects.